src/test_bubble.py

```diff
@@ -0,0 +1,41 @@
+"""Test bubble sort."""
+
+
+import pytest
+from random import randint
+
+
+to_sort = [
+    ([1, 2, 3, 4, 5, 6, 7, 8, 9, 10], [1, 2, 3, 4, 5, 6, 7, 8, 9, 10]),
+    ([10, 9, 8, 7, 6, 5, 4, 3, 2, 1], [1, 2, 3, 4, 5, 6, 7, 8, 9, 10]),
+    ([10, 9, 8, 17, 6, 300, 4, 3, 2, 11], [2, 3, 4, 6, 8, 9, 10, 11, 17, 300])
+]
+
+
+def test_bubble_non_list_raises_error():
+    """Entering a non-list/tuple param raises an error."""
+    from bubble import bubble_sort
+    with pytest.raises(TypeError):
+        bubble_sort('Hello')
+
+
+def test_bubble_non_int_raises_error():
+    """Entering an iterable containing non-integers raises an error."""
+    from bubble import bubble_sort
+    with pytest.raises(ValueError):
+        bubble_sort([1, 2, 3, 5, 'burp'])
+
+
+@pytest.mark.parametrize('input, expected', to_sort)
+def test_bubble_sort_returns_ordered_list(input, expected):
+    """Bubble sort returns an ordered list."""
+    from bubble import bubble_sort
+    assert bubble_sort(input) == expected
```

Comment:
You should avoid using "input" as a variable name as that's a reserved word in Python